packages/quantic/force-app/main/default/lwc/quanticResultHighlightedTextField/__tests__/quanticResultHighlightedTextField.js

```diff
@@ -0,0 +1,211 @@
+};
+
+const selectors = {
+  labelSelector: '.result-text__label',
+  valueSelector: '.result-text__value',
+  errorSelector: '.error-message',
+  highlightedSection: 'b',
+};
+
+const defaultOptions = {
+  result: exampleResult,
+  field: exampleField,
+};
+
+let isInitialized = false;
+
+// @ts-ignore
+mockHeadlessLoader.initializeWithHeadless = (element, engineId, initialize) => {
+  if (element instanceof QuanticResultHighlightedTextField) {
+    if (!isInitialized) {
+      isInitialized = true;
+      initialize();
+    }
+  }
+};
+
+// @ts-ignore
+mockHeadlessLoader.getHeadlessBundle = () => {
+  return {
+    ResultTemplatesHelpers: {
+      getResultProperty: (result, field) => result[`${field}`],
+    },
+    HighlightUtils: {
+      highlightString: () => exampleHighlightedFieldValue,
+    },
+  };
+};
+
+function createTestComponent(options = defaultOptions) {
+  const element = createElement(componentName, {
+    is: QuanticResultHighlightedTextField,
+  });
+
+  for (const [key, value] of Object.entries(options)) {
+    element[key] = value;
+  }
+
+  document.body.appendChild(element);
+  return element;
+}
+
+// Helper function to wait until the microtask queue is empty.
+function flushPromises() {
+  return new Promise((resolve) => setTimeout(resolve, 0));
+}
+
+describe('c-quantic-result-highlighted-text-field', () => {
+  function cleanup() {
+    // The jsdom instance is shared across test cases in a single file so reset the DOM
+    while (document.body.firstChild) {
+      document.body.removeChild(document.body.firstChild);
+    }
+    isInitialized = false;
+    jest.clearAllMocks();
+  }
+
+  beforeEach(() => {
+    console.error = jest.fn();
+    // @ts-ignore
+    global.Bueno = {
+      isString: jest
+        .fn()
+        .mockImplementation(
+          (value) => Object.prototype.toString.call(value) === '[object String]'
+        ),
+    };
+  });
+
+  afterEach(() => {
+    cleanup();
+  });
+
+  describe('when no result is given', () => {
+    it('should show an error', async () => {
+      // @ts-ignore
+      const element = createTestComponent({field: exampleField});
+      await flushPromises();
+
+      const errorMessage = element.shadowRoot.querySelector(
+        selectors.errorSelector
+      );
+
+      expect(errorMessage).not.toBeNull();
+      expect(console.error).toHaveBeenCalledWith(
+        `The ${componentName} requires a result and a field to be specified.`
+      );
+    });
+  });
+
+  describe('when no field is given', () => {
+    it('should show an error', async () => {
+      // @ts-ignore
+      const element = createTestComponent({result: exampleResult});
+      await flushPromises();
+
+      const errorMessage = element.shadowRoot.querySelector(
+        selectors.errorSelector
+      );
+
+      expect(errorMessage).not.toBeNull();
+      expect(console.error).toHaveBeenCalledWith(
+        `The ${componentName} requires a result and a field to be specified.`
```

Comment:
You could put that string as a const declaration at the beginning of your describe since you re-use three times

```const expectedConsoleErrorMessage = `The ${componentName} requires a result and a field to be specified.`;```